app/logical/danbooru/paginator/elasticsearch_extensions.rb

```diff
@@ -162,6 +162,16 @@ def total_count
 
         results.total
       end
+
+      def exists?
+        search.definition.update(from: 0, size: 0, terminate_after: 1)
```

Comment:
I've found that to actually get the fastest results for an existence query, you actually want these options:
`{ sort: '_doc', _source: false, size: 1, track_total_hits: false, terminate_after: 1 }`